test/itkCLEImageTest.cxx

```diff
@@ -0,0 +1,181 @@
+
+  diff->UpdateLargestPossibleRegion();
+
+  bool differenceFailed = false;
+
+  const double averageIntensityDifference = diff->GetTotalDifference();
+
+  const unsigned long numberOfPixelsWithDifferences = diff->GetNumberOfPixelsWithDifferences();
+
+  if (averageIntensityDifference > 0.0)
+  {
+    if (static_cast<int>(numberOfPixelsWithDifferences) > toleranceNumberOfPixels)
+    {
+      differenceFailed = true;
+    }
+    else
+    {
+      differenceFailed = false;
+    }
+  }
+  else
+  {
+    differenceFailed = false;
+  }
+
+  return differenceFailed;
+}
+
+int itkCLEImageTest(int argc, char* argv[])
+{
+  if (argc > 1)
+  {
+    std::cout << "received " << argc << " arguments, but didn't expect any."
+      << "first ignored argument: " << argv[1] << std::endl;
+  }
+  unsigned int width, height;
+  using ImageType = itk::CLEImage<float, 2>;
+
+  auto cleImage = ImageType::New();
+  ITK_EXERCISE_BASIC_OBJECT_METHODS(cleImage, CLEImage, Image)
+
+  std::cout << "Print: " << std::endl;
+  cleImage->Print(std::cout);
+
+  // set size & region
+  width = 256;
+  height = 256;
+  ImageType::IndexType start;
+  start[0] = 0;
+  start[1] = 0;
+  ImageType::SizeType size;
+  size[0] = width;
+  size[1] = height;
+  ImageType::RegionType region;
+  region.SetSize(size);
+  region.SetIndex(start);
+  ImageType::IndexType idx;
+  idx[0] = 0;
+  idx[1] = 0;
+  // setup our images
+  auto srcA = ImageType::New();
+  auto srcB = ImageType::New();
+  srcB->DebugOn();
+  srcA->SetRegions(region);
+  srcA->Allocate();
+  srcB->SetRegions(region);
+  srcB->Allocate();
+  double A, B; // Temp testing variables
+
+  // Ensure graft works
+  std::cout << "Test Graft." << std::endl;
+  cleImage->Graft(srcA);
+  if (cleImage->GetPixelContainer() != cleImage->GetPixelContainer())
+  {
+    std::cerr << "Graft test failed." << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  // Testing FillBuffer
+  srcA->FillBuffer(1.0f);
+  srcB->FillBuffer(1.0f);
+  if (Compare<ImageType>(cleImage,srcB))
+    return EXIT_FAILURE;
+
+  // Testing SetPixel
+  srcA->SetPixel(idx, 7.0f);
+  srcB->SetPixel(idx, 7.0f);
+  A = srcA->GetPixel(idx);
+  B = srcB->GetPixel(idx);
+  ITK_TEST_EXPECT_EQUAL(A, B);
+  if (Compare<ImageType>(srcA, srcB))
+    return EXIT_FAILURE;
+  A = srcA->GetPixel(idx);
+  B = srcB->GetPixel(idx);
+  ITK_TEST_EXPECT_EQUAL(A, B);
+
+  // Testing operator[]
+  auto pixA = (*srcA)[idx];
+  auto pixB = (*srcB)[idx];
+  pixA = 22.0f;
+  pixB = 22.0f;
+  A = srcA->GetPixel(idx);
+  B = srcB->GetPixel(idx);
+  ITK_TEST_EXPECT_EQUAL(A, B);
+  if (Compare<ImageType>(srcA, srcB))
+    return EXIT_FAILURE;
+  A = srcA->GetPixel(idx);
+  B = srcB->GetPixel(idx);
+  ITK_TEST_EXPECT_EQUAL(A, B);
+
+  // Testing GetBufferPointer
+  auto bufptrA = srcA->GetBufferPointer();
+  auto bufptrB = srcB->GetBufferPointer();
+  bufptrA[5] = 12.0f;
+  bufptrB[5] = 12.0f;
+  A = srcA->GetPixel(idx);
+  B = srcB->GetPixel(idx);
+  ITK_TEST_EXPECT_EQUAL(A, B);
```

Comment:
In these tests, `B` should explicitly be `12.0f` in the case, etc. `A` and `B` could both be `0.0` if setting the values failed and the test would still pass.